MusicTaste/MusicTaste/MatchingAlgorithm.m

```diff
@@ -58,35 +54,36 @@ + (double) compareSpotifyData:(PFObject *)potentialMatch withData:(NSManagedObje
                 NSInteger albumCount = [matchDataAlbums count];
                 NSInteger trackCount = [matchDataTracks count];
                 NSInteger artistCount = [matchDataArtists count];
+                //Makes copies of array to avoid deletion of data
+                NSMutableArray *matchDataGenresCopy = [[NSMutableArray alloc] initWithArray:matchDataGenres];
+                NSMutableArray *matchDataTracksCopy = [[NSMutableArray alloc] initWithArray:matchDataTracks];
```

Comment:
Created copies of the Arrays so I wouldn't delete the original array's information